lib/util.js

```diff
@@ -360,29 +368,31 @@ module.exports = {
   getSystemErrorName,
   inherits,
   inspect,
-  isArray: ArrayIsArray,
-  isBoolean,
-  isBuffer,
+  isArray: deprecate(ArrayIsArray,
+                     'The `util.isArray API is deprecated.',
+                     'DEP0044'),
+  isBoolean: deprecate(isBoolean, 'The `util.isBoolean API is deprecated.', 'DEP0045'),
+  isBuffer: deprecate(isBuffer, 'The `util.isBuffer API is deprecated.', 'DEP0046'),
   isDeepStrictEqual(a, b) {
     if (internalDeepEqual === undefined) {
       internalDeepEqual = require('internal/util/comparisons')
         .isDeepStrictEqual;
     }
     return internalDeepEqual(a, b);
   },
-  isNull,
-  isNullOrUndefined,
-  isNumber,
-  isString,
-  isSymbol,
-  isUndefined,
-  isRegExp: types.isRegExp,
-  isObject,
-  isDate: types.isDate,
-  isError,
-  isFunction,
-  isPrimitive,
-  log,
+  isNull: deprecate(isNull, 'The `util.isNull API is deprecated.', 'DEP0050'),
+  isNullOrUndefined: deprecate(isNullOrUndefined, 'The `util.isNullOrUndefined API is deprecated.', 'DEP0051'),
+  isNumber: deprecate(isNumber, 'The `util.isNumber API is deprecated.', 'DEP0052'),
+  isString: deprecate(isString, 'The `util.isString API is deprecated.', 'DEP0056'),
+  isSymbol: deprecate(isSymbol, 'The `util.isSymbol API is deprecated.', 'DEP0057'),
+  isUndefined: deprecate(isUndefined, 'The `util.isUndefined API is deprecated.', 'DEP0058'),
```

Comment:
```suggestion
  isUndefined: deprecate(
    isUndefined,
    'The `util.isUndefined API is deprecated. Please use `arg === undefined` instead.',
    'DEP0058'
  ),
```

What about suggesting the alternatives in the deprecation message right away? That way it's most straight forward for users receiving the message.

Reply:
is it done for any other deprecation messages?

Reply:
Most of these utils are quite straight forward

Reply:
I think the error message and the documentation are quite unhelpful, we should give concrete suggestions to the alternatives. They are not straight forward enough for people who don't like to keep these JavaScript nitty-gritty in their brain (TBH even I don't know remember how to do many of these in JS off the top of my head other than the typeof ones).